### PR TITLE
bf: handle 404 error from source gracefully

### PR DESCRIPTION
**Pull request by jonathan-gramain**
don't update source metadata, just log and skip the entry

### CODE REVIEW

File: extensions/replication/queueProcessor/QueueProcessor.js [hunk 1]
```diff
@@ -539,6 +545,12 @@ class QueueProcessor {
                             error: err.description });
                 return done();
             }
+            if (err.ObjNotFound) {
+                log.info('replication skipped: ' +
```
Comment (philipyoo) [inline extensions/replication/queueProcessor/QueueProcessor.js]: Minor, but maybe use `log.error`?

Comment (jonathan-gramain) [inline extensions/replication/queueProcessor/QueueProcessor.js]: It's actually not an error, just a singular case of not replicating anything. (the user can create and remove an object version on the source before the queue populator has a chance to look at it, it's a normal situation).

Comment (jonathan-gramain) [inline extensions/replication/queueProcessor/QueueProcessor.js]: I meant, before the queue processor processes it

**Comment by jonathan-gramain** [comment]
@rahulreddy the PENDING status is written on each version, so if the version has been deleted, it used to have the PENDING status but not anymore?

On access denied, we just skip because we don't have the access rights to update the replication status (unless access denied is only for reading but not for writing, which would be weird).

**Comment by jonathan-gramain** [comment]
Or you're thinking about the master version replication status?

**Comment by rahulreddy** [comment]
> the PENDING status is written on each version, so if the version has been deleted, it used to have the PENDING status but not anymore?
On access denied, we just skip because we don't have the access rights to update the replication status (unless access denied is only for reading but not for writing, which would be weird).

I understand better now, if we cannot write to the versioned copy we skip.

**Pull request merged**